floris/tools/layout_functions.py

```diff
@@ -397,18 +344,232 @@ def put_label(i, ax):
         if i < 0:  # sanitize negative i
             i = len(x) + i
-        put_label(i, ax)
+        put_label(i)
     elif near_x is not None:
         for i in range(len(x) - 2):
-            if (x[i] < near_x and x[i + 1] >= near_x) or (
-                x[i + 1] < near_x and x[i] >= near_x
-            ):
-                put_label(i, ax)
+            if (x[i] < near_x and x[i + 1] >= near_x) or (x[i + 1] < near_x and x[i] >= near_x):
+                put_label(i)
     elif near_y is not None:
         for i in range(len(y) - 2):
-            if (y[i] < near_y and y[i + 1] >= near_y) or (
-                y[i + 1] < near_y and y[i] >= near_y
-            ):
-                put_label(i, ax)
+            if (y[i] < near_y and y[i + 1] >= near_y) or (y[i + 1] < near_y and y[i] >= near_y):
+                put_label(i)
     else:
         raise ValueError("Need one of near_i, near_x, near_y")
+
+
+def plot_waking_directions(
+    fi: FlorisInterface,
+    ax: plt.Axes = None,
+    turbine_indices: List[int] = None,
+    wake_plotting_dict: Dict[str, Any] = {},
+    D: float = None,
+    limit_dist_D: float = None,
+    limit_dist_m: float = None,
+    limit_num: int = None,
+    wake_label_size: int = 7,
+) -> plt.Axes:
+    """
+    Plots lines representing potential waking directions between wind turbines in a layout.
+
+    Args:
+        fi (FlorisInterface): Instantiated FlorisInterface object containing layout data.
+        ax (plt.Axes, optional): An existing axes object to plot on. If None, a new
+            figure and axes will be created. Defaults to None.
+        turbine_indices (List[int], optional):  Indices of turbines to include in the plot.
+            If None, all turbines are plotted. Defaults to None.
+        wake_plotting_dict (Dict[str, Any], optional): Dictionary to customize the appearance
+            of waking direction lines. Valid keys include:
+                * 'color' (str): Line color. Defaults to 'black'.
+                * 'linestyle' (str): Line style (e.g., 'solid', 'dashed'). Defaults to 'solid'.
+                * 'linewidth' (float): Line width. Defaults to 0.5.
+        D (float, optional):  Rotor diameter. Used for distance calculations if `limit_dist_D`
+            is provided. If None, defaults to the first turbine's rotor diameter.
+        limit_dist_D (float, optional): Maximum distance between turbines (in rotor diameters)
+            to plot waking lines. Defaults to None (no limit).
+        limit_dist_m (float, optional): Maximum distance (in meters) between turbines to plot
+            waking lines. Overrides `limit_dist_D` if provided. Defaults to None (no limit).
+        limit_num (int, optional):  Limits the number of waking lines plotted from each turbine
+            to the `limit_num` closest neighbors. Defaults to None (no limit).
+        wake_label_size (int, optional): Font size for labels showing wake distance and direction.
+            Defaults to 7.
+
```

Comment:
The other docstrings nicely add the exceptions raised, so it might be good to add that here.

Reply:
Done!